test/sequential/crowdloan.API.claiming.test.ts

```diff
@@ -0,0 +1,329 @@
+/*
+ *
+ * @group crowdloan
+ *
+ */
+import { jest } from "@jest/globals";
+import { ApiPromise, Keyring } from "@polkadot/api";
+import { getApi, initApi } from "../../utils/api";
+import { Assets } from "../../utils/Assets";
+import { BN, hexToBn } from "@polkadot/util";
+import { setupApi, setupUsers } from "../../utils/setup";
+import { Sudo } from "../../utils/sudo";
+import { User } from "../../utils/User";
+import {
+  claimCrowdloanRewards,
+  completeCrowdloanInitialization,
+  initializeCrowdloanReward,
+  setCrowdloanAllocation,
+  sudoClaimCrowdloanRewards,
+} from "../../utils/tx";
+import {
+  getBlockNumber,
+  getEnvironmentRequiredVars,
+  waitBlockNumber,
+} from "../../utils/utils";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import { BN_ZERO, signTx } from "@mangata-finance/sdk";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+import { ExtrinsicResult, waitNewBlock } from "../../utils/eventListeners";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser1: User;
+let testUser2: User;
+let testUser3: User;
+let testUser4: User;
+let testUser5: User;
+let testUser6: User;
+let testUser7: User;
+let testUser8: User;
+let testUser9: User;
+let leaseStartBlock: number;
+let leaseEndingBlock: number;
+let api: ApiPromise;
+let sudo: User;
+let keyring: Keyring;
+let crowdloanId: any;
+const crowdloanRewardsAmount = new BN("1000000000000000000000000");
+const nativeCurrencyId = MGA_ASSET_ID;
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  api = getApi();
+  sudo = new User(keyring, sudoUserName);
+
+  await setupApi();
+
+  [testUser1, testUser2, testUser3, testUser4, testUser5] = setupUsers();
+  [testUser6, testUser7, testUser8, testUser9] = setupUsers();
+
+  testUser1.addAsset(nativeCurrencyId);
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(sudo),
+    Assets.mintNative(testUser1),
+    Assets.mintNative(testUser2),
+    Assets.mintNative(testUser3),
+    Assets.mintNative(testUser4),
+    Assets.mintNative(testUser5),
+    Assets.mintNative(testUser6),
+    Assets.mintNative(testUser7),
+    Assets.mintNative(testUser8),
+    Assets.mintNative(testUser9)
+  );
+
+  await setCrowdloanAllocation(crowdloanRewardsAmount.muln(8));
+
+  await Sudo.batchAsSudoFinalized(
```

Comment:
if you rewrite the initialize method to accept a list of users, we can simplify this.

Reply:
Done